### PR TITLE
Use tighter limits on FCU water flow rate if RegulaFalsi iteration limit is exceeded

### PR DESCRIPTION
**Pull request by rraustad**
## Pull request overview

Very low load can cause fan coil unit to reach maximum iterations while searching for a solution. When this occurs, a new function creates tighter limits on water flow rate so that the solution can converge. This fixes defect #5612. No diff's in current example files are expected.
### Work Checklist

Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
- [x] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
- [x] At least one of the following appropriate labels must be added to this PR to be consumed into the changelog:
  - Defect: This pull request repairs a github defect issue.  The github issue should be referenced in the PR description
### Review Checklist

This will not be exhaustively relevant to every PR.
- [x] Code style (parentheses padding, variable names)
- [x] Functional code review (it has to work!)
- [x] If defect, results of running current develop vs this branch should exhibit the fix
- [x] CI status: all green or justified
- [x] Performance: CI Linux results include performance check -- verify this
- [x] Unit Test(s)
- C++ checks:
  - [x] Argument types
  - [x] If any virtual classes, ensure virtual destructor included, other things
- IDD changes n/a

### CODE REVIEW

**Comment by rraustad** [comment]
This issue is resolved and CI should return all green.

File: src/EnergyPlus/FanCoilUnits.cc [hunk 1]
```diff
@@ -2095,28 +2105,39 @@ namespace FanCoilUnits {
...
+							} else if ( SolFlag == -2 ) {
+								++FanCoil( FanCoilNum ).LimitErrCountH;
+								if ( FanCoil( FanCoilNum ).LimitErrCountH < 2 ) {
+									ShowWarningError( "Hot Water control failed in fan coil unit " + FanCoil( FanCoilNum ).Name );
+									ShowContinueError( "  Bad hot water mass flow limits" );
+									ShowContinueErrorTimeStamp( "..Water flow rate set to lower limit " );
+								} else {
+									ShowRecurringWarningErrorAtEnd( "Hot Water control failed in fan coil unit " + FanCoil( FanCoilNum ).Name,
+										FanCoil( FanCoilNum ).BadMassFlowLimIndexH );
+								}
```
Comment (rraustad) [inline src/EnergyPlus/FanCoilUnits.cc:2147]: Where's the other } else if ( SolFlag == -2 ) ? This looks wrong. See line 2027.

**Comment by rraustad** [comment]
Correct RegulaFalsi test for failure. Defect file only runs in release mode due to this code. These asserts show when a surface is ill-formed. However, there is no indication of which surface and this surface still runs in release mode. These should be deleted.

```
#ifndef NDEBUG // Check x_bot and x_top are also sorted
                Real64 xb( std::get< 0 >( crossEdges[ 0 ] ) );
                Real64 xt( std::get< 1 >( crossEdges[ 0 ] ) );
                Real64 const tol( 1.0e-9 * std::max( std::abs( xl ), std::abs( xu ) ) ); // EnergyPlus vertex precision is not tight so tolerance isn't either
                for ( auto const & edge: crossEdges ) { // Detect non-simple polygon with crossing edges
                    Real64 const xbe( std::get< 0 >( edge ) );
                    Real64 const xte( std::get< 1 >( edge ) );
                    assert( xb <= xbe + tol );
                    assert( xt <= xte + tol );
                    xb = xbe;
                    xt = xte;
                }
#endif
```

**Comment by rraustad** [comment]
Defect file only runs in release mode and runs without "Cold water control failed" warnings in Chicago and Brisbane.

**Comment by mjwitte** [comment]
@rraustad While this solves the issue, it concerns me that the fancoil controls have to reach the max iteration limit first, then reset limits and try again.  This could be quite slow.  Any chance this condition can be trapped before the first round of iterations?  Say, anytime the load is <10% of nominal capacity then tighten limits first. Or lower the iteration limit which is currently 100!  @wfbuhl Any thoughts?

Also, there are some missing spaces on lines 2026, 2028, 2032, 2149, 2151.

**Comment by rraustad** [comment]
I wrestled with the thought of tightening limits all the time and then considered the files that are solved in only a few iterations. Maybe the suggestion of testing for 10% load avoids additional iterations for those times when a solution would be found using only a few iterations.

**Comment by mjwitte** [comment]
So what's the typical number of iterations needed?  If it typically can converge in a few iterations, then maybe the initial iteration limit should be something like 10 or 15, and then tighten limits and let that one go to 25?  I have a set of files from a recent helpdesk ticket (11373) that exhibit a large number of fancoil iteration errors.  I'll add them to this DefectFiles folder.  The -P33 file has no fancoil iteration errors, -P66 has a few, and -P100 has a bunch more.  Runtime tripled between -P33 and -P100.  These files may have other issues, but there should be some way for the fancoil to come to a solution quickly even if it can't meet the load or is thrashing around to nail a tiny load.  Is the convergence tolerance for tiny loads appropriate?

**Comment by wfbuhl** [comment]
Well, I guess since you fixed it I can't look at the issue. Typically, 10
or less iterations. Obviously not a good solution scheme, but that may not
be the issue. Can you send me the issue?

**Comment by mjwitte** [comment]
@wfbuhl Any insights to offer on this one?

**Comment by rraustad** [comment]
One other thing I was thinking about. If the water flow needed to meet the load is > 10%, then calling into this new routine is only 1 additional iteration and may not add much overhead, might even save that on the solution iterations. I wonder what the difference is in speed if I just called this every time?

**Comment by nrel-bot-3** [comment]
@mjwitte @rraustad @lgentile it has been 14 days since this pull request was last updated.

**Comment by mjwitte** [comment]
I've tested this with the original defect file and it runs without any fancoil iteration warnings.  The additional files that were referenced [above](https://github.com/NREL/EnergyPlus/pull/5671#issuecomment-223581978) actually exhibit a _different_ fancoil warning so I'm posting a new issue for those.

**Pull request merged**